nntrainer/include/tensor.h

```diff
@@ -491,20 +539,26 @@ class Tensor {
    * @brief     return current stride of tensor.
    * @retval    int[MAXDIM] strides
    */
-  const int *strides() const noexcept { return _strides; }
+  const std::array<int, MAXDIM> strides() const noexcept { return _strides; }
 
 private:
-  /**< handle the data as a std::vector type */
-  std::vector<float> data;
-  int _strides[MAXDIM];
-  bool _is_contiguous;
+  /**
+   * @brief     return Data pointer of Tensor
+   * @retval    float pointer
+   */
+  float *getData() { return _data.get(); }
+
+  /**< handle the data as a std::shared_ptr<float> type */
   TensorDim dim;
+  std::array<int, MAXDIM> _strides;
```

Comment:
any reason to use `_` as a prefix to all these variables?

Reply:
It is easier to tell private member variable from public member variables as well as adding some kind of awareness that you are using private variable.

AFAIK, variable name policy hasn't been discussed w.r.t. this matter and open to opinion about this :)

Reply:
let's do not use _. looks more complicated.

Reply:
Updated.